app/services/open_provider_courses.rb

```diff
@@ -6,7 +6,12 @@ def initialize(provider:)
   end
 
   def call
-    if run_courses.or(ratified_courses).update_all(open_on_apply: true).positive?
+    courses_to_open = run_courses.or(ratified_courses)
```

Comment:
What are your thoughts around using `run_courses.or(ratified_courses).update(name: nil, open_on_apply: true)` instead, which does run callbacks/validations, to avoid having to update each course in turn?

Reply:
Ah, thanks! Did not realise that was a thing!